keystore/src/transaction/mod.rs

```diff
@@ -341,12 +341,13 @@ impl KeystoreTransaction {
                 (identifier_12, E2eiRefreshToken),
                 (identifier_13, E2eiAcmeCA),
                 (identifier_14, E2eiIntermediateCert),
-                (identifier_15, E2eiCrl)
+                (identifier_15, E2eiCrl),
+                (identifier_16, ConsumerData)
             ],
             proteus_types: [
-                (identifier_16, ProteusPrekey),
-                (identifier_17, ProteusIdentity),
-                (identifier_18, ProteusSession)
+                (identifier_17, ProteusPrekey),
```

Comment:
I don't know how these identifiers are used. Are these identifiers persisted at all? Can anything go wrong by moving `identifier_16` from `ProteusPrekey` to `ConsumerData`?

Reply:
These identifiers are merely used to hold the data from the in-memory cache before it is committed. So it doesn't matter what their name is/previously was at all. 

If we can create them internally in the macro instead of passing them in, I would like doing so!

Reply:
Creating idents like this within a macro is tricky but not impossible. 

In the future, it will be easier: combine the [`${index()}` metafunction](https://github.com/rust-lang/rust/pull/122808) with the [`paste!` macro](https://crates.io/crates/paste) and we should be able to generate all these idents automatically. 

For now, `${index()}` is still not stable, so you have to use [recursive macro tricks](https://veykril.github.io/tlborm/decl-macros/building-blocks/counting.html#bit-twiddling) to build up a unary number for each index and then count how many symbols appear in that number. This approach is complicated and opaque enough I'd recommend not reworking the existing macro until `${index()}` stabilizes.